### ISSUE TITLE
shrinkwrap-cache: A local shrinkwrap caching daemon

### ISSUE THREAD

**Issue by Raynos** [Raynos/funpm#2]
To truly optimize a `git clone && funpm install` we could utilize a local shrinkwrap caching daemon.

Such a daemon would make any CI server or build server a lot faster.
- What it would do is take an `npm-shrinkwrap.json` file which must include all dev deps & checksums.
- It would split it into N npm-shrinkwrap.json files; one for every top level dependency. 
- It would then look at `cache/{shrinkwrapSha}/tarball.tar.gz` and return that if it exists
- If it does not exist; it would fetch all the deps from the shrinkwrap in parallel; build them; tar them up and store them in the cache; then return them

Such a caching daemon would allow us to install from an `npm-shrinkwrap.json` in an efficient manner. If you install the same shrinkwrap twice its instant (i.e. you added a new commit without changing deps and want to run your CI server again).

If you changed 1 or 2 deps out of 30 you would only have to fetch a small percentage of the modules and would be able to very aggressively grab everything else from the cache.
